README.md

```diff
@@ -16,3 +17,11 @@ Also give a theoretical argument for why X could or could not be correct, based
 on the complexity of the general sorting problem we covered in class.
 
 Add your answers to this markdown file.
+
+## Testing for $O(n)$
+
+If we only have a blackbox algorithm, then we'll have to test it and graph it. I would test it with different cases, like being sorted, slightly sorted, randomly shuffled, completely reversed for the worst case, and lists of the same number, and make about 30 different versions of each case, going from 1 to 30 length. I would also test empty lists. Then we can graph the time it took to sort over the change in length and find a linear regression line for the data. If the $r$ is significantly close to 1, then we do truly have a linear algorithm, otherwise the algorithm does not have a $O(n)$ complexity. 
```

Comment:
What is $r$?

Reply:
yes my bad. $r$ is the correlation coefficient, it would tell us how well the line of best fit conforms to our data. It ranges from -1 to 1, where -1 is negatively correlated, and 1 is completely fit or correlated. 0 is the middle spot where it is neither negatively correlated or positively correlated.

Reply:
Does that necessarily imply a linear correlation?

Reply:
It wouldn't if we just found a linear regression line for the data, cause that would be anything. I changed my analysis so that I picked my linear regression line to be $y=x$.

Reply:
Then if r was significantly close to 1 it would be linear.